stdlib/LinearAlgebra/src/tridiag.jl

```diff
@@ -170,6 +170,10 @@ adjoint(S::SymTridiagonal) = Adjoint(S)
 Base.copy(S::Adjoint{<:Any,<:SymTridiagonal}) = SymTridiagonal(map(x -> copy.(adjoint.(x)), (S.parent.dv, S.parent.ev))...)
 Base.copy(S::Transpose{<:Any,<:SymTridiagonal}) = SymTridiagonal(map(x -> copy.(transpose.(x)), (S.parent.dv, S.parent.ev))...)
 
+ishermitian(S::SymTridiagonal{<:Real}) = true
+ishermitian(S::SymTridiagonal{<:Complex}) = all(x -> imag(x) == 0, S.dv) && all(x -> imag(x) == 0, S.ev)
```

Comment:
```suggestion
ishermitian(S::SymTridiagonal) = isreal(S.dv) && isreal(S.ev)
```

Reply:
Clearer, but slightly slower (by ~50ns) on my machine with `S = SymTridiagonal(rand(100) .+ 0im, rand(100) .+ 0im)`.

Reply:
That's what `isreal` is good for, and we use it even internally in `LinearAlgebra` for that purpose.